### PR TITLE
Improve commit-name.sh

### PR DESCRIPTION
**Pull request by ghost**
Used https://www.shellcheck.net/ to check for mistakes and fixed them.

### CODE REVIEW

**Review by yuyichao — COMMENTED** [review]
Most of the quotes are not necessary but also looks harmless.

File: contrib/commit-name.sh [hunk 1]
```diff
@@ -7,37 +7,37 @@
...
+    verchanged=$(git blame -L ,1 -sl "$gitref" -- "$topdir/VERSION" | cut -f 1 -d " ")
     nb=$(git rev-list --count "$gitref" "^$verchanged")
-    pre=$(echo $ver | cut -s -f 2 -d "-")
-    if [ $ver = "0.5.0-dev" ]; then
+    pre=$(echo "$ver" | cut -s -f 2 -d "-")
+    if [ "$ver" = "0.5.0-dev" ]; then
         # bump to 0.5.0-dev was one commit after tag during 0.5.0-dev
-        nb=$(expr $nb + 1)
-    elif [ $ver = "0.5.0-pre" ]; then
+        nb=$((nb + 1))
```
Comment (yuyichao) [inline contrib/commit-name.sh]: At least dash accepts it.

Comment (albap) [inline contrib/commit-name.sh]: I don't have POSIX standard for shell to check, but this guys says it's POSIX http://stackoverflow.com/a/22389873

Comment (albap) [inline contrib/commit-name.sh]: Confirmed http://pubs.opengroup.org/onlinepubs/007904875/utilities/xcu_chap02.html#tag_02_06_04

**Comment by albap** [comment]
> Most of the quotes are not necessary but also looks harmless.

It prevents globbing and word splitting.
https://github.com/koalaman/shellcheck/wiki/SC2086

Even if it's not always necessary it's always a good idea to use quotes for failproofing and future changes.
Keep on the safe side.

**Comment by yuyichao** [comment]
> It prevents globbing and word splitting.

Yes, I know that. Most of them are hashes.

**Comment by KristofferC** [comment]
It seems we won't do this. Splitting similar type of changes into 8 separate PRs also makes this hard to review so another attempt at this would be better of in a single PR.

**Pull request closed without merging**